sockeye/training.py

```diff
@@ -586,7 +588,27 @@ def fit(self,
                 for name, val in metric_val.get_name_value():
                     logger.info('Checkpoint [%d]\tValidation-%s=%f', self.state.checkpoint, name, val)
 
-                # (3) update training metrics
+                # (2) determine stopping
+                if 0 <= max_num_not_improved <= self.state.num_not_improved:
+                    logger.info("Maximum number of not improved checkpoints (%d) reached: %d",
+                                max_num_not_improved, self.state.num_not_improved)
+                    self.state.converged = True
+
+                    if min_epochs is not None and self.state.epoch < min_epochs:
+                        logger.info("Minimum number of epochs (%d) not reached yet: %d",
+                                    min_epochs, self.state.epoch)
+                        self.state.converged = False
+
+                    if min_updates is not None and self.state.updates < min_updates:
+                        logger.info("Minimum number of updates (%d) not reached yet: %d",
+                                    min_updates, self.state.updates)
+                        self.state.converged = False
+
+                    if min_samples is not None and self.state.samples < min_samples:
+                        logger.info("Minimum number of samples (%d) not reached yet: %d",
+                                    min_samples, self.state.samples)
```

Comment:
is there actually a `self.state.converged = False` missing?

Reply:
Strange. looks like it's was missing all along, in the previous code. Adding in the next commit.